src/test/scala/com/cognite/sdk/scala/v1/TransformationsTest.scala

```diff
@@ -0,0 +1,214 @@
+
+      val resByExternalIds = client.transformations.retrieveByExternalIds(externalIds, true)
+      resByExternalIds.size shouldBe transformationsToCreate.size
+    } finally
+      try
+        client.transformations.delete(externalIds.map(CogniteExternalId(_)), true)
+      catch {
+        case NonFatal(_) => // ignore
+      }
+  }
+
+  it should "filter transformations" in {
+
+    val transformationsToCreate = (0 to 3).map { i =>
+      TransformationCreate(
+        s"transformation-sdk-test-$i",
+        Some(s"select abc $i"),
+        Some(GeneralDataSource("events")),
+        conflictMode = Some("upsert"),
+        Some(false),
+        sourceOidcCredentials = Some(credentials.toFlatCredentials),
+        destinationOidcCredentials = Some(credentials.toFlatCredentials),
+        externalId = shortRandomUUID(),
+        ignoreNullFields = true,
+        dataSetId = Some(216250735038513L)
+      )
+    }
+    val resCreates = client.transformations.createItems(Items(transformationsToCreate))
+    val externalIds = transformationsToCreate.map(_.externalId)
+
+    try {
+      val resFilterName = client.transformations
+        .filter(TransformationsFilter(nameRegex = Some("transformation-sdk")))
+        .compile
+        .toList
+      resFilterName.size shouldBe transformationsToCreate.size
+      resCreates.map(_.name) should contain theSameElementsAs resFilterName.map(_.name)
+
+      val resFilterQueryRegex = client.transformations
+        .filter(TransformationsFilter(queryRegex = Some("select abc")))
+        .compile
+        .toList
+      resFilterQueryRegex.size >= transformationsToCreate.size shouldBe true
+      resFilterQueryRegex.map(_.query).forall(q => q.startsWith("select abc")) shouldBe true
+
+      val resFilterUpsertEvents = client.transformations
+        .filter(
+          TransformationsFilter(destinationType = Some("events"), conflictMode = Some("upsert"))
+        )
+        .compile
+        .toList
+      resFilterUpsertEvents.size >= transformationsToCreate.size shouldBe true
+      resFilterUpsertEvents
+        .map(_.destination.asInstanceOf[GeneralDataSource].`type`)
+        .toSet shouldBe Set("events")
+      resFilterUpsertEvents.map(_.conflictMode).toSet shouldBe Set("upsert")
+
+      val max1DaysAgo = TimeFilter(max = Some(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli))
+      val resFilterCreatedTime1DayAgo = client.transformations
+        .filter(
+          TransformationsFilter(
+            nameRegex = Some("transformation-sdk"),
+            createdTime = Some(max1DaysAgo)
+          )
+        )
+        .compile
+        .toList
+      resFilterCreatedTime1DayAgo.isEmpty shouldBe true
+
+      val min3MinsAgo =
+        TimeFilter(min = Some(Instant.now().minus(3, ChronoUnit.MINUTES).toEpochMilli))
+      val resFilterCreatedTime3MinsAgo = client.transformations
+        .filter(
+          TransformationsFilter(
+            nameRegex = Some("transformation-sdk"),
```

Comment:
This too should use a unique prefix name for this test run, so other tests don't interfere with it.

Reply:
You did, but you're not using it here, and I think you should. ;)

Reply:
right, fixed :)